subsys/random/CMakeLists.txt

```diff
@@ -7,6 +7,12 @@ zephyr_library()
 zephyr_library_sources_ifdef(CONFIG_USERSPACE           rand32_handlers.c)
 endif()
 
+if (CONFIG_TEST_RANDOM_GENERATOR)
+  message(WARNING "This application is using a not truly random generator."
+    "This capability is not secure and it is provided for testing purposes only."
+    "Use it carefully.")
+endif()
```

Comment:
I think you should print the actual CONFIG_TEST_RANDOM_GENERATOR name in the message instead of forcing them to search the source tree for the message. Tell the user that the named CONFIG item is not intended for production applications.

Also, I found that formating the message() a little differently will allow the actual message to be highlighted in the build output. See #28027 as an example I was pulling together.

Reply:
I think I've got it. Could you please review it again ?